core/src/main/java/org/elasticsearch/indices/ttl/IndicesTTLService.java

```diff
@@ -99,7 +99,7 @@ protected void doStop() {
         try {
             this.purgerThread.shutdown();
         } catch (InterruptedException e) {
-            Thread.interrupted();
+            Thread.currentThread().interrupt();
```

Comment:
This is called in `#doStop()` which is called when the component is about to stop anyway and does no harm.